datadog_checks_dev/datadog_checks/dev/tooling/testing.py

```diff
@@ -58,16 +67,28 @@ def get_tox_envs(checks, style=False, format_style=False, benchmark=False, every
         yield check, envs_selected
 
 
-def get_available_tox_envs(check, sort=False, e2e_only=False):
-    if e2e_only:
+def get_available_tox_envs(check, sort=False, e2e_only=False, e2e_tests_only=False):
+    if e2e_tests_only:
+        tox_command = 'tox --listenvs-all -v'
+    elif e2e_only:
         tox_command = 'tox --listenvs-all'
     else:
         tox_command = 'tox --listenvs'
 
     with chdir(path_join(get_root(), check)):
-        env_list = run_command(tox_command, capture='out').stdout
+        output = run_command(tox_command, capture='out').stdout
+
+    env_list = [e.strip() for e in output.splitlines()]
 
-    env_list = [e.strip() for e in env_list.splitlines()]
+    if e2e_tests_only:
+        envs = []
+        for line in env_list:
+            if '->' in line:
+                env, _, description = line.partition('->')
+                if 'e2e ready' in description.lower():
```

Comment:
Why do you think we need something like that? We need to write the tests, so it's not like it would load tests that are not ready?

Reply:
As stated in the PR description this is for tox envs, not tests themselves

Reply:
Oh yeah that's a good point. We don't really have a way to make the env change/enforce the agent python version though?

Reply:
Python 3 is not yet shipped with Agent, no